todo/src/App.js

```diff
@@ -52,4 +29,4 @@ const mapStateToProps = (state) => {
     };
 };
 
-export default connect(mapStateToProps, { addTodo, toggleTodo, clearCompleted })(App);
+export default connect(mapStateToProps, { clearCompleted })(App);
```

Comment:
much better, now App only is aware of the acitoncreator it needs to make button work and not the others. keep in mind that since you still don't need any of the store's state either, you don't even need to spread it to your props here with mapStateToProps, and you can pass in `null` for that instead: `export default connect(null, { clearCompleted })(App);`